File: CCIP/CCIP/UIApplication+addition.m
//
//  UIApplication+addition.m
//  CCIP
//
//  Created by 腹黒い茶 on 2016/08/01.
//  Copyright © 2016年 CPRTeam. All rights reserved.
//

#import "UIApplication+addition.h"

@implementation UIApplication (addition)

+ (UIViewController *)getMostTopPresentedViewController {
    UIViewController *vc = [[[UIApplication sharedApplication] keyWindow] rootViewController];
    while ([vc presentedViewController])
        vc = [vc presentedViewController];
    return vc;
}

@end

File: CCIP/CCIP/Class+addition/UIApplication+addition.swift
//
//  UIApplication+addition.swift
//  OPass
//
//  Created by 腹黒い茶 on 2018/11/4.
//  Copyright © 2018 OPass. All rights reserved.
//

import Foundation

@objc extension UIApplication {
    static func getMostTopPresentedViewController() -> UIViewController? {
        var vc : UIViewController? = UIApplication.shared.keyWindow?.rootViewController;
        while (vc!.presentedViewController != nil) {
            vc = vc!.presentedViewController;
        }
        return vc;
    }
}
